newrelic/core/attribute.py

```diff
@@ -174,6 +175,35 @@ def resolve_agent_attributes(attr_dict, attribute_filter, target_destination, at
     return a_attrs
 
 
+def resolve_logging_context_attributes(attr_dict, attribute_filter, attr_prefix, attr_class=dict):
+    """
+    Helper function for processing logging context attributes that require a prefix. Correctly filters attribute names
+    before applying the required prefix, and then applies the process_user_attribute after the prefix is applied to
+    correctly check length requirements.
+    """
+    c_attrs = attr_class()
+
+    for attr_name, attr_value in attr_dict.items():
+        if attr_value is None:
```

Comment:
If we are adding user attributes to a transaction, we would capture an attribute {"foo": None} on the transaction as {"foo": "None"}. Here however, we drop the transaction instead. It seems like these should behave the same in both scenarios?

Reply:
This matches the implementation in resolve_user_attributes which is used by spans and trace segments, unsure why the behavior is different between these. I'm not super tied to one or the other, but this matches most closely to these other resolve functions from this file.

Reply:
span attributes actually behave the same way. The attributes actually appear to be processed twice which is different than logging user attributes and seems to be why the None is slipping past that check on spans and transactions. They are processed originally [here](https://github.com/newrelic/newrelic-python-agent/blob/main/newrelic/core/node_mixin.py#L23) which converts the None to the string "None" and then again the resolve_user_attributes function. I'm not actually sure why there's a None check in there at all, since it doesn't seem like that would be reachable in the code. Either way, if you just take out the None check in this new function I think the behavior will be consistent across all the data types. I don't really see the point in processing them twice-I would think this would actually be a bad thing for performance. It seems like a mistake to me, but this code is complex so I could be missing something.

Reply:
Sorry got a little confused there when I was explaining that ^. It looks like the original function doesn't process the attributes again so they are only processed once but the point is that they are processed before the resolve_ function is called so the None gets converted to a "None" string.

Reply:
That feels like it might be a bug but there's nothing in the spec either way that I can see so let's stick with sending it up then. I changed the PR.

Reply:
I'm inclined to agree with you on that but for now, I think it should be consistently wrong rather than a mix of the two depending on the datatype.